src/operations/blas3/gemm_local_tensorcore.hpp

```diff
@@ -0,0 +1,814 @@
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gemm_local_tensorcore.hpp
+ *
+ **************************************************************************/
+
+#ifndef SYCL_BLAS_BLAS3_LOCAL_GEMM_TENSORCORE_HPP
+#define SYCL_BLAS_BLAS3_LOCAL_GEMM_TENSORCORE_HPP
+
+#include "gemm_common.hpp"
+#include "gemm_load_store.hpp"
+
+namespace blas {
+
+/*!
+ * @brief GemmFactory is a template class whose instantiations provide
+ *        different implementations of the GEMM device function.
+ *
+ * To use the function, each item of a kernel launched with nd_range given by
+ * GemmFactory::get_nd_range() should call GemmFactory::run(). The size of
+ * local memory required per work group can be queried with
+ * GemmFactory::local_memory.
+ *
+ * @tparam DoubleBuffer  iff true,  enables the use of double buffering
+ *                       (doubles the amount of consumed local memory,
+ *                        but halves the number of required local barriers)
+ * @tparam NbcA  iff true, avoids bank conflicts when accessing blocks of
+ *               matrix A in local memory (slightly increases local
+ *               memory consumption) - may be useful in combination with TranA
+ * @tparam NbcA  iff true, avoids bank conflicts when accessing blocks of
+ *               matrix B in local memory (slightly increases local
+ *               memory consumption) - may be useful in combination with TranB
+ * @tparam ClSize  the size of the cache line of the architecture
+ *                 (If the value passed in is smaller than the actual cache
+ *                 line size, some values fetched will be wasted, which can
+ *                 significantly reduce performance. It can be set to a
+ *                 multiple of the physical cache line size. In this case, it
+ *                 will significantly increase local memory usage, but
+ *                 will result in fewer local barriers.)
+ * @tparam TileType  determines the size of the local, work group, and top
+ *                   level tiles to use, see Tile
+ * @tparam TransA  iff true, matrix A will be transposed on the fly
+ * @tparam TransB  iff true, matrix B will be transposed on the fly
+ * @tparam element_t  type of matrix elements
+ * @tparam is_beta_zero True if beta == 0.
+ * @tparam VectorSize The packet size to be used for vectorization.
+ * @tparam batch_type the type of batch strideded /interleaved
+ * @tparam UseTensorcores boolean parameter to decide whether to use Tensorcores
+ * or not
+ */
```

Comment:
I think the iff either looks like a misspelling or unnecessary cleverness.

Reply:
iff is used in math for "if and only if", but I am not sure if most developers are familiar with this.

Reply:
Resolving this since it is - for better or worse - consistent with other parts of SYCL-BLAS.